conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,642 @@
+        }
+        template = Template(_jinja_format_list_values() + self.template, trim_blocks=True, lstrip_blocks=True, undefined=StrictUndefined)
+        return template.render(context)
+
+
+class DepContentGenerator:
+    """
+    Generates Makefile content for a dependency
+    """
+
+    template = textwrap.dedent("""\
+
+        # {{ dep.ref.name }}/{{ dep.ref.version }}
+
+
+        CONAN_NAME_{{ name }} = {{ dep.ref.name }}
+
+        CONAN_VERSION_{{ name }} = {{ dep.ref.version }}
+
+        CONAN_ROOT_{{ name }} = {{ root }}
+
+        {{  define_variable_value("CONAN_SYSROOT_{}".format(name), sysroot) -}}
+        {{- define_variable_value("CONAN_INCLUDE_DIRS_{}".format(name), include_dirs) -}}
+        {{- define_variable_value("CONAN_LIB_DIRS_{}".format(name), lib_dirs) -}}
+        {{- define_variable_value("CONAN_BIN_DIRS_{}".format(name), bin_dirs) -}}
+        {{- define_variable_value("CONAN_SRC_DIRS_{}".format(name), src_dirs) -}}
+        {{- define_variable_value("CONAN_BUILD_DIRS_{}".format(name), build_dirs) -}}
+        {{- define_variable_value("CONAN_RES_DIRS_{}".format(name), res_dirs) -}}
+        {{- define_variable_value("CONAN_FRAMEWORK_DIRS_{}".format(name), framework_dirs) -}}
+        {{- define_variable_value("CONAN_OBJECTS_{}".format(name), objects) -}}
+        {{- define_variable_value("CONAN_LIBS_{}".format(name), libs) -}}
+        {{- define_variable_value("CONAN_DEFINES_{}".format(name), defines) -}}
+        {{- define_variable_value("CONAN_CFLAGS_{}".format(name), cflags) -}}
+        {{- define_variable_value("CONAN_CXXFLAGS_{}".format(name), cxxflags) -}}
+        {{- define_variable_value("CONAN_SHAREDLINKFLAGS_{}".format(name), sharedlinkflags) -}}
+        {{- define_variable_value("CONAN_EXELINKFLAGS_{}".format(name), exelinkflags) -}}
+        {{- define_variable_value("CONAN_FRAMEWORKS_{}".format(name), frameworks) -}}
+        {{- define_variable_value("CONAN_REQUIRES_{}".format(name), requires) -}}
+        {{- define_variable_value("CONAN_SYSTEM_LIBS_{}".format(name), system_libs) -}}
+        {{- define_variable_value("CONAN_COMPONENTS_{}".format(name), components) -}}
+        """)
+
+    def __init__(self, dependency: object, root: str, sysroot: str, dirs: dict, flags: dict):
+        self._dep = dependency
+        self._root = root
+        self._sysroot = sysroot
+        self._dirs = dirs or {}
+        self._flags = flags or {}
+
+    def content(self) -> str:
+        """
+        Parse dependency variables and generate its Makefile content
+        """
+        context = {
+            "dep": self._dep,
+            "name": _makefy(self._dep.ref.name),
+            "root": self._root,
+            "sysroot": self._sysroot,
+            "include_dirs": self._dirs.get("include_dirs", []),
+            "lib_dirs": self._dirs.get("lib_dirs", []),
+            "bin_dirs": self._dirs.get("bin_dirs", []),
+            "src_dirs": self._dirs.get("src_dirs", []),
```

Comment:
Isn't it possible to use the ``None`` in the rendering as an empty indicator, instead of having to default to all empty lists?